var/spack/repos/builtin/packages/py-cupy/package.py

```diff
@@ -31,7 +33,10 @@ class PyCupy(PythonPackage, CudaPackage):
     depends_on("py-fastrlock@0.5:", type=("build", "run"))
     depends_on("py-numpy@1.20:1.25", when="@:11", type=("build", "run"))
     depends_on("py-numpy@1.20:1.26", when="@12:", type=("build", "run"))
-    depends_on("py-scipy@1.6:1.12", type=("build", "run"))
+
+    depends_on("py-scipy@1.6:1.12", when="+all", type=("build", "run"))
```

Comment:
Missing deps on cython and optuna. Alternatively, you can get rid of the variant and deps.

Reply:
I'll just remove it for now. We can break out the all variant in future if desired.